### ISSUE TITLE
Filename always ignores "Title" setting

### ISSUE THREAD

**Issue by RapGameDerrida** [GitJournal/GitJournal#541]
**Operating system**: iOS 14.8 and iPadOS 15 public beta

**Issue**: the filename for new notes appears to always ignore the "Title" option for "New Note Filename". Instead, a new note will default to date, or whatever option was used before switching to Title (eg ISO8601). As such, anyone who wants to have notes named after the title field needs to manually edit the file name every time.

**Reproduce**: 

1. Go to Settings
2. Under "Storage", change New Note Filename to anything *but* Title. In this case, yyyy-mm-dd
3. Create a new note, check "edit file name". yyyy-mm-dd works fine.
4. Return to settings, change New Note Filename to Title.
5. Create a new note, add title (or not), check "edit file name". The file will still be saved as yyyy-mm-dd, as if the Title setting didn't stick.

Screenshots attached in order, from settings to creating a new note to the filename ignoring the setting.

![64594A28-E7CE-4D00-8181-206E05F8EA9D](https://user-images.githubusercontent.com/72739872/133978296-a0eb51a7-1cd4-4154-bfa7-6ab1920ae858.jpeg)
![23074A12-7212-4BAD-9232-27A9D6E7020D](https://user-images.githubusercontent.com/72739872/133978302-387e2741-66bd-4c8b-bdc3-833e63ce85ad.jpeg)
![51F36DC5-B471-4F5A-8449-E7208029E223](https://user-images.githubusercontent.com/72739872/133978303-c71579b3-d996-446e-8d13-2b3af64ef4dc.jpeg)

**Comment by vHanda** [comment]
I just tried this and I can't seem to reproduce it. The code also looks correct :(

Maybe it's an ios specific bug (unlikely though), I've tried it out on Android so far. I'm just releasing version `1.80.4`, could you please let me know if this still occurs for you? If so, could you try restarting the app after each time you change the settings, and attach the debug log please. (Settings -> Debug)

**Comment by brwarner** [comment]
Hey, just wanted to pipe in as an Android user that I'm also seeing this. I can try `1.80.4` as well with a restart.

**Comment by RapGameDerrida** [comment]
@vHanda

1.84 not there yet (unsurprisingly, no worries there), but I updated from 1.80 to 1.80.2. Also upgraded to iOS 15. Good news: the Title setting behaves as expected! I tried changing between different filename options, then back to Title, and everything works just fine.

Not sure if @brwarner was also on 1.80 (except on Android), and if .80.2 or .4 will solve it—so perhaps not _totally_ solved yet.

Just in case, here's the debug log. Includes both 1.80 and 1.80.2, in case comparing the two makes something jump out. (Let me know if something other than a plain text copy paste is needed.)

[debug.txt](https://github.com/GitJournal/GitJournal/files/7205774/debug.txt)
